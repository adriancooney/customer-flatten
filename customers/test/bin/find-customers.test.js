const util = require("../../src/util");
const findCustomers = require("../../bin/find-customers");

describe("find-customers", () => {
    let logMock, fatalMock;

    beforeAll(() => {
        logMock = jest.spyOn(util, "log");
        fatalMock = jest.spyOn(util, "fatal");
    });

    beforeEach(() => {
        logMock.mockReset();
        fatalMock.mockReset();
    });

    afterAll(() => {
        logMock.mockRestore();
        fatalMock.mockRestore();
    });

    it("should output help if `--help` is passed", async () => {
        await findCustomers(["--help"]);

        expect(fatalMock).toHaveBeenCalled();
    });

    it("should output help if no arguments passed", async () => {
        await findCustomers([]);

        expect(fatalMock).toHaveBeenCalled();
    });

    it("should correctly read in customer data and output customers", async () => {
        const processMock = jest.spyOn(process, "cwd").mockImplementation(() => __dirname);

        await findCustomers(["../../data/customers.log"]);

        processMock.mockRestore();

        expect(logMock).toHaveBeenCalled();
        expect(logMock.mock.calls[0][0]).toMatchSnapshot();
    });
});
