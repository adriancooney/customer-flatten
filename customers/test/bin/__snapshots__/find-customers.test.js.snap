// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`find-customers should correctly read in customer data and output customers 1`] = `
"name,user_id
\\"Ian Kehoe\\",4
\\"Nora Dempsey\\",5
\\"Theresa Enright\\",6
\\"Eoin Ahearn\\",8
\\"Richard Finnegan\\",11
\\"Christina McArdle\\",12
\\"Olive Ahearn\\",13
\\"Michael Ahearn\\",15
\\"Patricia Cahill\\",17
\\"Eoin Gallagher\\",23
\\"Rose Enright\\",24
\\"Stephen McArdle\\",26
\\"Oliver Ahearn\\",29
\\"Nick Enright\\",30
\\"Alan Behan\\",31
\\"Lisa Ahearn\\",39"
`;
